src/krb5_asrep_fmt_plug.c

```diff
@@ -110,30 +142,58 @@ static size_t cracked_size;
 static int new_keys;
 
 static struct custom_salt {
-	dyna_salt dsalt;
-	unsigned char edata1[16];
 	uint32_t edata2len;
-	unsigned char* edata2;
+	unsigned char edata1[16];
+	char salt[256];
+	unsigned char edata2[4096]; // XXX is this large enough?
+	unsigned char etype;
```

Comment:
This is a bad change. You should keep using dyna_salt - that means there is no "large enough" questions. Salt will be as large as needed.